engine/gui/src/gui.cpp

```diff
@@ -2065,8 +2065,8 @@ Result DeleteDynamicTexture(HScene scene, const dmhash_t texture_hash)
                             }
                         }
 
-                        n->m_ParentIndex = INVALID_INDEX;
                         ResetInternalNode(scene, n);
+                        n->m_ParentIndex = INVALID_INDEX;
```

Comment:
The internals of ResetInternalNode does a RemoveFromList, which requires it to know the parent, in order to remove it from the parent's list of children.